src/optimizer/query_rewrite.c

```diff
@@ -6906,6 +7391,12 @@ qo_optimize_queries (PARSER_CONTEXT * parser, PT_NODE * node, void *arg, int *co
 	      single_tuple_bak = node->info.query.flag.single_tuple;
 	      node->info.query.flag.single_tuple = false;
 
+	      /* order by, union check */
+	      if (node->info.query.order_by == NULL && !qo_check_distinct_union (parser, node)
+		  && !node->info.query.q.select.hint & PT_HINT_NO_PUSH_PRED)
```

Comment:
Shouldn't it be modified as follows in consideration of operator precedence?

 if (node->info.query.order_by == NULL && !qo_check_distinct_union (parser, node)
		  && **!(node->info.query.q.select.hint & PT_HINT_NO_PUSH_PRED)**)

Reply:
It is better to modify it considering operator precedence.

Reply:
I'll fix it